### PR TITLE
microcode_test: Add case to test microcode version is passed to guest

### PR DESCRIPTION
**Pull request by huangyum**
ID: 1886664
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

**Comment by huangyum** [comment]
@yanan-fu Please help review, thanks.

Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.microcode_test: PASS (52.63 s)

**Comment by huangyum** [comment]
@vivianQizhu Would you pls help review, thanks.

File: qemu/tests/microcode_test.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    def get_microcode_ver(cmd, session=None):
+        """
+        Get microcde version in guest or host
+        """
+        if session:
+            output = session.cmd_output(cmd)
+        else:
+            output = process.getoutput(cmd, shell=True)
+        ver = re.findall(r":\s?(0x[0-9A-Fa-f]+)", output)[0]
```
Comment (yanan-fu) [inline qemu/tests/microcode_test.py:28]: How about `r":\s*(0x[0-9A-Fa-f]+` ? replace `?` with `*`, to cover 0, 1 and more space.

Comment (huangyum) [reply qemu/tests/microcode_test.py:28]: It seems not right with a single '0x' but no any hexadecimal digits.

Comment (huangyum) [reply qemu/tests/microcode_test.py]: Oh, I got you. Updated.

**Review by yanan-fu — APPROVED** [review]
Acked-by: Yanan Fu <yfu@redhat.com>

File: qemu/tests/microcode_test.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    try:
+        error_context.context("Get microcode version in guest", logging.info)
+        guest_ver = get_microcode_ver(cmd)
+        logging.info("The microcode version in guest is %s", guest_ver)
+        if guest_ver != host_ver:
+            test.fail("The microcode version in guest does not match host")
+    finally:
+        session.close()
```
Comment (vivianQizhu) [inline qemu/tests/microcode_test.py:46]: Probably session exits in case that guest hits exceptions and is down during the test, recommend: `if session:`.

**Review by vivianQizhu — COMMENTED** [review]
LGTM otherwise.

File: qemu/tests/microcode_test.py (hunk 2 above)
Comment (huangyum) [reply qemu/tests/microcode_test.py]: Thanks, updated.

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**